### PR TITLE
use html base for single source of truth

### PR DESCRIPTION
**Pull request by Anton-4**
`ROC_DOCS_URL_ROOT` was previously used to put "hardcoded" links in many href's, now we put the root in the HTML base tag and use relative links everywhere. I already used this for the basic-cli 0.3.2 docs.

### CODE REVIEW

File: crates/docs/src/lib.rs [hunk 1]
```diff
@@ -101,23 +102,24 @@ pub fn generate_docs_html(root_file: PathBuf) {
...
+        .replace("<!-- styles.css -->", "styles.css")
         .replace("<!-- favicon.svg -->", "/favicon.svg")
         .replace(
             "<!-- Prefetch links -->",
             loaded_module
                 .docs_by_module
                 .iter()
                 .map(|(_, module)| {
-                    let href = module_link_url(module.name.as_str());
+                    let href = module.name.as_str();
```
Comment (Anton-4) [inline crates/docs/src/lib.rs:114]: use a relative link instead

File: crates/docs/src/lib.rs [hunk 2]
```diff
@@ -256,12 +254,12 @@ fn render_module_documentation(
                 if all_exposed_symbols.contains(&doc_def.symbol) {
                     buf.push_str("<section>");
 
-                    let name = doc_def.name.as_str();
-                    let href = format!("#{name}");
+                    let def_name = doc_def.name.as_str();
```
Comment (Anton-4) [inline crates/docs/src/lib.rs:257]: Just a more descriptive name.

File: crates/docs/src/lib.rs (hunk 1 above)
Comment (rtfeldman) [reply crates/docs/src/lib.rs:113]: Hm, I think these started out as relative and then got changed to root...but I don't remember why! I'll try to dig it up...

Comment (Anton-4) [reply crates/docs/src/lib.rs:113]: @rtfeldman should we just merge this and see if any issues pop up? It's been working well for the basic-cli 0.3.2 docs

Comment (rtfeldman) [reply crates/docs/src/lib.rs:113]: Sure, that works 👍

Comment (rtfeldman) [reply crates/docs/src/lib.rs:113]: @Anton-4 ok turns out it breaks all the CSS if you refresh 😂

Comment (Anton-4) [reply crates/docs/src/lib.rs:113]: fixed :)

**Pull request merged**